docs/geometry.rst

```diff
@@ -173,3 +173,17 @@ specified, rounding precision will be disabled showing full precision.
 Format types ``'x'`` and ``'X'`` show a hex-encoded string representation of
 WKB or Well-Known Binary, with the case of the output matched the
 case of the format type character.
+
+.. _canonical-form:
```

Comment:
I might be wrong about this, but I think canonical form may also impose some convention as to which point in a polygon comes first? Or perhaps it is just the shapely implementation that does this. Anyway, if so, this should also be noted and explained. 

As I have said, I might be wrong, but e.g. (note that p1 is in CW order) so this is not a result of a winding order reversal:
```
>>> import shapely
>>> p1 = shapely.geometry.Polygon([(0, 0), (0, 100), (100, 100), (100, 0)])
>>> p2 = shapely.set_precision(p1, grid_size = 1e-3)
>>> print(f"{p1=}\n{p2=}")
p1=<POLYGON ((0 0, 0 100, 100 100, 100 0, 0 0))>
p2=<POLYGON ((0 100, 100 100, 100 0, 0 0, 0 100))>
```
This behaviour might be less likely to bother people, but it's still a behaviour!

Reply:
@DOSull the second point is the following:
* the starting point of rings is lower left.

This isn't clear?

Reply:
Hmm... good point.

I did some extra tests... and apparently the result of `set_precision` isn't completely in canonical form... that specific aspect is not correcly applied...

Not sure what to do with it... Strictly speaking it is probably a bug in geos... but....

```
import shapely

p1 = shapely.geometry.Polygon([(0, 0), (100, 0), (100, 100), (0, 100)])
p2 = shapely.set_precision(p1, grid_size=1e-3)
p3 = shapely.normalize(p1)
p4 = shapely.normalize(p2)

print(f"{p1=} (counter-clockwise)")
print(f"{p2=} (clockwise)")
print(f"{p3=} (clockwise)")
print(f"{p4=} (clockwise)")
```

Output:
```
p1=<POLYGON ((0 0, 100 0, 100 100, 0 100, 0 0))> (counter-clockwise)
p2=<POLYGON ((0 100, 100 100, 100 0, 0 0, 0 100))> (clockwise)
p3=<POLYGON ((0 0, 0 100, 100 100, 100 0, 0 0))> (clockwise)
p4=<POLYGON ((0 0, 0 100, 100 100, 100 0, 0 0))> (clockwise)
```

Reply:
@DOSull I did some more tests and updated the documentation accordingly.